plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1900,10 +1901,13 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
         if (!fullyDeletedFiles.isEmpty()) {
             try {
                 TrinoFileSystem fileSystem = fileSystemFactory.create(session);
-                fileSystem.deleteFiles(fullyDeletedFiles.values().stream()
+                List<String> filesToDelete = fullyDeletedFiles.values().stream()
                         .flatMap(Collection::stream)
                         .map(CommitTaskData::getPath)
-                        .collect(toImmutableSet()));
+                        .collect(toImmutableList());
+                for (List<String> batch : Lists.partition(filesToDelete, DELETE_BATCH_SIZE)) {
+                    fileSystem.deleteFiles(batch);
```

Comment:
What's the benefit?
if the FS needs batches, why not let it do on its own?

Reply:
i agree with https://github.com/trinodb/trino/pull/14908/commits/2c87d6b56584d704cde26c8b2dd84b70f5373064#r1015558983 that the data list is already in memory (in some form)